packages/react/src/jsx/ReactJSXElement.js

```diff
@@ -966,6 +966,11 @@ export function cloneAndReplaceKey(oldElement, newKey) {
     __DEV__ && enableOwnerStacks ? oldElement._debugStack : undefined,
     __DEV__ && enableOwnerStacks ? oldElement._debugTask : undefined,
   );
+  if (__DEV__) {
+    // The cloned element should inherit the original element's key validation.
+    clonedElement._store.validated = oldElement._store.validated;
```

Comment:
Do we need to check if `oldElement._store` exists? We do so in other calls.

Reply:
I noticed that, too.

I don't think it's necessary because we always define it in `__DEV__`, and we also assume it exists shortly after this method call (when we conditionally set `validated` to 2).

Reply:
Perhaps the other call sites check for `_store` because they might be receiving malformed elements (e.g. constructed by third party libraries or something).

But here, we are creating `clonedElement`.